bin/hdfcoinc/pycbc_foreground_censor

```diff
@@ -28,26 +28,61 @@ pycbc.init_logging(args.verbose)
-start = f['segments/foreground_veto/start'][:]
-end = f['segments/foreground_veto/end'][:]
-vsegs = pycbc.events.start_end_to_segments(start, end)
-
-ifo1, ifo2 = f.attrs['detector_1'], f.attrs['detector_2']
-
-ifos, fsegs, names = [], [], []
-for ifo in [ifo1, ifo2]:
-    segs = pycbc.events.select_segments_by_definer(args.veto_file, args.segment_name, ifo)
-    fsegs += [(segs.coalesce() + vsegs.coalesce()).coalesce()]
-    names += [args.output_segment_name]
-    ifos += [ifo]
-
-if args.strict_coinc_time:
-    fsegs[0] = (fsegs[0].coalesce() + fsegs[1].coalesce()).coalesce()
-    fsegs[1] = fsegs[0]
-
-# FIXME: Don't use URL once Larne supports taking file paths directly in the
-#        workflow File class
-file_url = urlparse.urlunparse(['file', 'localhost',
+
+if 'segments/foreground_veto/start' in f:
+    start = f['segments/foreground_veto/start'][:]
+    end = f['segments/foreground_veto/end'][:]
+    vsegs = pycbc.events.start_end_to_segments(start, end)
+
+    ifo1, ifo2 = f.attrs['detector_1'], f.attrs['detector_2']
+
+    ifos, fsegs, names = [], [], []
+    for ifo in [ifo1, ifo2]:
+        segs = pycbc.events.select_segments_by_definer(args.veto_file, args.segment_name, ifo)
+        fsegs += [(segs.coalesce() + vsegs.coalesce()).coalesce()]
+        names += [args.output_segment_name]
+        ifos += [ifo]
+
+    if args.strict_coinc_time:
+        fsegs[0] = (fsegs[0].coalesce() + fsegs[1].coalesce()).coalesce()
+        fsegs[1] = fsegs[0]
+
+    # FIXME: Don't use URL once Larne supports taking file paths directly in the
+    #        workflow File class
+    file_url = urlparse.urlunparse(['file', 'localhost',
                           os.path.abspath(args.output_file), None, None, None])
-SegFile.from_multi_segment_list('UNUSED', fsegs, names, ifos, file_url=file_url)
-logging.info('done')
+    SegFile.from_multi_segment_list('UNUSED', fsegs, names, ifos, file_url=file_url)
+else:
+    ifolist = f.attrs['ifos'].split(' ')
+    ifokey = ''.join(sorted(ifolist))
+
+    start = f['segments/%s/foreground_veto/start' % ifokey][:]
+    end = f['segments/%s/foreground_veto/end' % ifokey][:]
+    mvsegs = pycbc.events.start_end_to_segments(start, end)
+
+    for key in f['segments']:
+        start = f['segments/%s/foreground_veto/start' % key][:]
+        end = f['segments/%s/foreground_veto/end' % key][:]
+        vsegs = pycbc.events.start_end_to_segments(start, end)
+        mvsegs = (mvsegs & vsegs).coalesce()
+
+    ifos, fsegs, names = [], [], []
+    for ifo in ifolist:
+        segs = pycbc.events.select_segments_by_definer(args.veto_file, args.segment_name, ifo)
+        fsegs += [(segs.coalesce() + mvsegs.coalesce()).coalesce()]
+        names += [args.output_segment_name]
+        ifos += [ifo]
+
+    if args.strict_coinc_time:
+        combfseg = fsegs[0]
+        for fseg in fsegs:
+            combfseg = (combfseg & fseg).coalesce()
+
+        for fseg in fsegs:
+            fseg = combfseg
```

Comment:
This is very confusing to read and understand straight away - can you add some comments to explain what is happening (what the if case is here)

Also for legibility, rename `for fseg in fsegs` to `for fseg_ifo in fsegs` (or similar) to make it so that it is easily identifiable as fseg for each ifo

Reply:
The strict_coinc_time option is wrong here for the multiifo case, so I removed it alltogether.